app/test/testMetadata.js

```diff
@@ -0,0 +1,181 @@
+  icon_url: 'http://www.example.com/rich-icon.png',
+  image_url: 'http://www.example.com/preview.png',
+  images: [
+    {
+      'url': 'http://www.example.com/preview.png',
+      'entropy': 1,
+      'height': 500,
+      'width': 500,
+    }
+  ],
+  original_url: 'http://www.example.com/good',
+  provider_url: 'http://www.example.com/good',
+  title: 'An Example Page',
+  url: 'http://www.example.com/good'
+};
+
+const goodExampleUrl = 'http://www.example.com/good';
+const badExampleUrl = 'http://www.example.com/bad';
+
+
+describe('Metadata API Tests', function() {
+  afterEach(function() {
+    fetchMock.restore();
+  });
+
+  it('should raise 415 if content header is not set', (done) => {
+    chai.request(app)
+      .post('/')
+      .send(JSON.stringify({urls: [goodExampleUrl]}))
+      .end((err, res) => {
+        res.should.have.status(415);
+        res.body.error.should.equal(errorMessages.headerRequired);
+        done();
+      });
+  });
+
+  it('should raise 400 if missing JSON body', (done) => {
+    chai.request(app)
+      .post('/')
+      .set('content-type', 'application/json')
+      .end((err, res) => {
+        res.should.have.status(400);
+        res.body.error.should.equal(errorMessages.urlsRequired);
+        done();
+      });
+  });
+
+  it('should raise 400 if missing urls parameter', (done) => {
+    chai.request(app)
+      .post('/')
+      .set('content-type', 'application/json')
+      .send(JSON.stringify({}))
+      .end((err, res) => {
+        res.should.have.status(400);
+        res.body.error.should.equal(errorMessages.urlsRequired);
+        done();
+      });
+  });
+
+  it('should raise 400 if urls is empty', (done) => {
+    chai.request(app)
+      .post('/')
+      .set('content-type', 'application/json')
+      .send(JSON.stringify({urls: []}))
+      .end((err, res) => {
+        res.should.have.status(400);
+        res.body.error.should.equal(errorMessages.urlsRequired);
+        done();
+      });
+  });
+
+  it('should return 200 with metadata for a url', (done) => {
+    fetchMock.mock(
+      goodExampleUrl,
+      `
+        <html>
+          <head>
+            <title>${exampleMetadata.title}</title>
+            <meta name="description" content="${exampleMetadata.description}" />
+            <meta name="thumbnail" content="${exampleMetadata.images[0].url}" />
+            <link rel="icon" href="http://www.example.com/rich-icon.png" />
+          </head>
+          <body></body>
+        </html>
+      `
+    );
+
+    chai.request(app)
+      .post('/')
+      .set('content-type', 'application/json')
+      .send(JSON.stringify({urls: [goodExampleUrl]}))
+      .end((err, res) => {
+
+        res.should.have.status(200);
+
+        const expectedResponse = {
+          error: '',
+          urls: {}
+        };
+        expectedResponse.urls[goodExampleUrl] = exampleMetadata;
+
+        res.body.should.deep.equal(expectedResponse);
```

Comment:
I think this could generally be refactored to this ES6-dynamic-array-key-gem:

```js
      .end((err, res) => {
        const expectedResponse = {
          error: '',
          urls: {
            [goodExampleUrl]: expectedMetadata
          }
        };

        res.should.have.status(200);
        res.body.should.deep.equal(expectedResponse);
```